pulp_smash/tests/rpm/api_v2/test_schedule_consumer.py

```diff
@@ -0,0 +1,163 @@
+# coding=utf-8
+"""Test that verify scheduled RPM installation and un-installation."""
+import datetime
+import random
+import time
+import unittest
+from urllib.parse import urljoin
+
+from pulp_smash import api, cli, config, exceptions, selectors, utils
+from pulp_smash.constants import (
+    CONSUMERS_PATH,
+    REPOSITORY_PATH,
+    RPM_UNSIGNED_FEED_URL,
+)
+from pulp_smash.tests.rpm.api_v2.utils import gen_distributor, gen_repo
+from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+REQUIRED_SERVICES = ('goferd',)
+
+REQUIRED_PACKAGES = (
+    'pulp-agent',
+    'pulp-consumer-client',
+    'pulp-rpm-consumer-extensions',
+    'pulp-rpm-handlers',
+    'pulp-rpm-yumplugins',
+    'python-gofer-qpid',
+)
+
+
+class ScheduledTaskConsumer(unittest.TestCase):
+    """Verify scheduled RPM installation and un-installation work.
+
+    Test the installation and un-installation of scheduled package in a
+    consumer.
+    This test case targets:
+    * `Pulp #2680 <https://pulp.plan.io/issues/2680>`_
+    * `Pulp Smash #611 <https://github.com/PulpQE/pulp-smash/issues/611>`_
+    """
+
+    def setUp(self):
+        """Install required packages and start the ``goferd`` service.
+
+        This is done because ``goferd`` must be running in order to communicate
+        with ``pulp-consumer-client``.
+        """
+        self.cfg = config.get_config()
+        if selectors.bug_is_untestable(2680, self.cfg.version):
+            self.skipTest('https://pulp.plan.io/issues/2680')
+        self.pkg_not_installed = []
+        cli_client = cli.Client(self.cfg)
```

Comment:
This line of code makes a CLI client that accesses the first system listed in `self.cfg`. Yet a couple lines down, a specific `system` is selected with `self.cfg.get_systems('shell')[0]`, and that `system` is used when initializing `self.svc_mgr`. It's very easy to imagine a scenario where the first system in `self.cfg` is different from the first system returned by `self.cfg.get_systems(...)`.